src/particles.jl

```
@@ -306,10 +304,22 @@ for PT in (:Particles, :StaticParticles, :WeightedParticles)
         Base.:\(p::Vector{<:$PT}, p2::Vector{<:$PT}) = Matrix(p)\Matrix(p2) # Must be here to be most specific
     end
 
+    @eval Base.promote_rule(::Type{S}, ::Type{$PT{T,N}}) where {S<:Number,T,N} = $PT{promote_type(S,T),N}
+    @eval Base.promote_rule(::Type{Bool}, ::Type{$PT{T,N}}) where {T,N} = $PT{promote_type(Bool,T),N}
 
-
+    for PT2 in (:Particles, :StaticParticles, :WeightedParticles)
+        if PT == PT2
+            @eval Base.promote_rule(::Type{$PT{S,N}}, ::Type{$PT{T,N}}) where {S,T,N} = $PT{promote_type(S,T),N}
+        elseif any(==(:StaticParticles), (PT, PT2))
+            @eval Base.promote_rule(::Type{$PT{S,N}}, ::Type{$PT2{T,N}}) where {S,T,N} = StaticParticles{promote_type(S,T),N}
+        else
+            @eval Base.promote_rule(::Type{$PT{S,N}}, ::Type{$PT2{T,N}}) where {S,T,N} = Particles{promote_type(S,T),N}
+        end
+    end
 end
 
+Base.promote_rule(::Type{<:AbstractParticles{S,M}}, ::Type{<:AbstractParticles{T,N}}) where {S,T,M,N} = Union{}
```

I'm not sure I understand this one, should there ever be a situation where this promote rule is called? The above rules should cover all situations, unless the user explicitly calls this method?

---

The problem is that `AbstractParticles <: Number`, so otherwise that method would get called, which leads to the problem described above. I think it's better to throw an error here instead